### ISSUE TITLE
Reuse of Synapse_JsonParser in Python instead of Azure function

### ISSUE THREAD

**Issue by plaksnor** [microsoft/DataLineage#3]
Hi,
I'm very excited about this SparkLin project! Great work!
I'm running Apache Spark 3.3 with a custom Apache Spark configuration with the [openlineage-spark-0.29.2.jar](https://repo1.maven.org/maven2/io/openlineage/openlineage-spark/0.29.2/openlineage-spark-0.29.2.jar) package. Since we're not allowed to use Azure functions or Kafka for this organization I'm working for, I use 'console' as the transport type. This way I get CustomTransport events in the console (Apache Spark log: stderr). These stderr files can be analyzed or parsed for further processing.

So I'm trying to parse the OpenLineage events by processing the stderr logs directly from within my Synapse Notebook. This approach has a benefit for other notebook users in the future, because this way they will get more flexibility by being able to choose themselves **when** they want to publish their entities and processes to Purview. We actually don't want **all** notebook logs to be parsed to Purview. Now users just simply need to embed 1 additional line of code by using a magic command (%run) which includes this notebook for publishing assets in Purview.

To make this work, I've been analyzing one of the functions like [Synapse_JsonParser.py](https://github.com/microsoft/DataLineage/blob/c0b6b287ed9d7e163786d8db0c2b8592c8a72e09/sparklin/BlobTriggerFuncApp/BlobTriggerFunction/Synapse_JsonParser.py). I'm able to parse the 'stderr' logs and see all kind of AtlasEntity and AtlasProcess calls showing up, so the concept works. Unfortunately, I don't get very good results from the jsonparser.
One thing to mention: I'm also not in the position to get a live connection with the Purview API. My goal is to create all the AtlasEntity and AtlasProcess calls to execute in another script later on. Not relevant for now.

For example, this...
```
import pandas as pd
pandasDF1 = pd.DataFrame({"customerId"   : [1, 2],
                         "customerCity" : ["Amsterdam", "Antwerpen"]})
```
...creates this (please ignore the GUID, because I don't have a live connection with the Purview API):
```
AtlasEntity(
   name=sparkDF1_view,
   typeName=DataSet,
   qualified_name=external://sparkdf1_view,
   guid=-100
)

AtlasProcess(
   name=external_sparkDF1_view_process,
   typeName=HRServicesInsights_OneHRSI,
   qualified_name=hrdi://synapse_notebook/nb_purview_automatisch_3/external_sparkDF1_view,
   inputs=[],
   outputs=[external://sparkdf1_view],
   attributes={'columnMapping': '[]', 'hardCoded_Columns': [], 'Delta_Tables': [], 'Global_Temp_Views_or_Tables': [], 'Intermediate_Views_or_Tables': ['sparkDF1_view'], 'JoinConditions': []},
   guid=-100
)
```

As you can see, it misses a lot of information. I really would like to get to the situation I could reuse Synapse_JsonParser.py logic to get the same results as what SparkLin is supposed to do.

So in short:
I'm looking for a way we could make a framework to parse OpenConsole events with Python like Synapse_JsonParser.py, which creates Atlas calls. These calls could be performed by another scripts which actually does have a live connection with Purview.

Would anyone be open for making this work?
Thanks in advance.

Attached you will find a stderr file with all the OpenLineage events as a ConsoleTransport type.
[stderr.log](https://github.com/microsoft/DataLineage/files/12130208/stderr.log)
